### ISSUE TITLE
fix(advection): public `reconstruct()` / `upwind_3pt` / `upwind_5pt` reconstruct the wrong face for u < 0 (off by exactly one cell; linear, WENO and WENO-Z)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#293]
Part of #276.

## Summary

In `finitevolx/_src/advection/reconstruct_fn.py` the negative-velocity branch of `upwind_3pt` (`:120-126`) and `upwind_5pt` (`:178-190`) reconstructs the face at `k + 1/2` while the positive branch reconstructs `k + 3/2`, so for `u < 0` every output is shifted by one cell. Through the public `reconstruct(q, u, dim, method, num_pts)` the error is exactly 1.0 on a unit ramp for all three flavours. The class path (`Reconstruction2D.weno5_x` etc.), which every public advection class uses, is correct; nothing internal calls `reconstruct`, and the only negative-flow test covers `upwind_1pt`.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp
from finitevolx import reconstruct
q = jnp.arange(12.0)[None, :].repeat(4, 0)            # cell averages of q = x, shape [Ny, Nx]
for method in ("linear", "weno", "wenoz"):
    for u in (1.0, -1.0):
        r = reconstruct(q, jnp.full(q.shape, u), dim=1, method=method, num_pts=5)
        print(method, u, r[0, 3:7])
```

```
upwind_3pt(linear) u<0 output[k] (expected k+1.5): [0.5 1.5 2.5 3.5 4.5]   u>0: [1.5 2.5 3.5 4.5 5.5]
upwind_5pt(linear) u<0 output[k] (expected k+2.5): [1.5 2.5 3.5 4.5 5.5]   u>0: [2.5 3.5 4.5 5.5 6.5]
reconstruct num_pts=3|5, method=linear|weno|wenoz: max err u>0 ≤ 1.8e-15,  u<0 = 1.00
Reconstruction2D.weno5_x u<0: 1.8e-15 (class path correct)
```

## Root cause

`reconstruct_fn.py:120-126, 178-190`: `h_neg = weno_*_right(q[..., :-2], q[..., 1:-1], q[..., 2:])` uses the stencil centred on cell k (face k+1/2) where `h_pos` is centred on k+1 (face k+3/2). `reconstruct_fn.py` also re-types the linear coefficients instead of importing `linear.py`.

## Proposed fix

Either shift the negative-branch slices by one (`q[..., 1:-1], q[..., 2:], q[..., 3:]`-style) so both branches target the same face and add a negative-flow ramp test, or delete `reconstruct_fn.py` and point `reconstruct` at the class path (`Reconstruction2D`), which is canonical and tested. Deleting is preferred: three parallel reconstruction pipelines is one of the structural findings (#280).

## Acceptance criteria

- [ ] Ramp test: `reconstruct(q=x, u=−1)` matches `reconstruct(q=x, u=+1)` at every face to 1e-14 for linear/weno/wenoz × num_pts 3/5.
- [ ] Mirror test: `reconstruct(flip(q), −u) == flip(reconstruct(q, u))`.
- [ ] Either `reconstruct_fn.py` is removed or its functions are used by the classes.
